x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportPutShutdownNodeAction.java

```diff
@@ -72,9 +70,16 @@ public ClusterState execute(ClusterState currentState) {
                 }
 
                 // Verify that there's not already a shutdown metadata for this node
-                if (Objects.nonNull(currentShutdownMetadata.getAllNodeMetadataMap().get(request.getNodeId()))) {
-                    logger.error(Strings.toString(currentShutdownMetadata));
-                    throw new IllegalArgumentException("node [" + request.getNodeId() + "] is already shutting down");
+                SingleNodeShutdownMetadata existingRecord = currentShutdownMetadata.getAllNodeMetadataMap().get(request.getNodeId());
+                if (existingRecord != null) {
+                    logger.trace(
+                        "replacing existing shutdown record for node [{}] of type [{}] with reason [{}] with new type [{}] and reason [{}]",
```

Comment:
I think this is a rare enough occurrence, and useful enough, to update to this:

```suggestion
                    logger.info(
                        "updating existing shutdown record for node [{}] of type [{}] with reason [{}] with new type [{}] and reason [{}]",
```